tests/src/test/java/alluxio/job/persist/PersistIntegrationTest.java

```diff
@@ -61,4 +65,50 @@ public void persistTest() throws Exception {
     final long jobId = mJobMaster.run(new PersistConfig("/test", 1, false, status.getUfsPath()));
     waitForJobFailure(jobId);
   }
+
+  @Test
+  @LocalAlluxioClusterResource.Config(
+      confParams = {
+          PropertyKey.Name.MASTER_PERSISTENCE_MAX_TOTAL_WAIT_TIME_MS, "1ms",
+          PropertyKey.Name.MASTER_PERSISTENCE_CHECKER_INTERVAL_MS, "50ms",
+          PropertyKey.Name.MASTER_PERSISTENCE_SCHEDULER_INTERVAL_MS, "50ms"})
+  public void persistTimeoutTest() throws Exception {
+    // write a file in alluxio only
+    AlluxioURI filePath = new AlluxioURI(TEST_URI);
+    FileOutStream os = mFileSystem.createFile(filePath,
+        CreateFileOptions.defaults().setWriteType(WriteType.MUST_CACHE));
+    os.write((byte) 0);
+    os.write((byte) 1);
+    os.close();
+    // check the file is completed but not persisted
+    URIStatus status = mFileSystem.getStatus(filePath);
+    Assert.assertEquals(PersistenceState.NOT_PERSISTED.toString(), status.getPersistenceState());
+    Assert.assertTrue(status.isCompleted());
+    // kill job worker
+    mLocalAlluxioJobCluster.getWorker().stop();
+    // persist the file
+    FileSystemContext context = FileSystemContext.get();
+    FileSystemMasterClient client = context.acquireMasterClient();
+    try {
+      client.scheduleAsyncPersist(new AlluxioURI(TEST_URI));
+    } finally {
+      context.releaseMasterClient(client);
+    }
+    // verify scheduled to be persisted
+    status = mFileSystem.getStatus(filePath);
+    Assert.assertEquals(PersistenceState.TO_BE_PERSISTED.toString(), status.getPersistenceState());
```

Comment:
this could be flaky across different machines. The file might already be in the `NOT_PERSISTED` stage if the call to `getStatus` takes too long. I suggest not testing the `TO_BE_PERSISTED` phase